contracts/DxMgnPool.sol

```diff
@@ -92,8 +111,16 @@ contract DxMgnPool {
             return (amount / totalDeposit) * totalPoolShares;
         }
     }
+    
+    function calculateClaimableMgn(Participation memory participation) private returns (uint) {
+        if (totalMgn == 0) {
+            totalMgn = mgnToken.balanceOf(address(this));
+        }
+        uint duration = auctionCount - participation.startAuctionCount;
+        return totalMgn * participation.poolShares * duration / totalPoolSharesCummulative;
```

Comment:
here we also need to divide by max-duration = auctionCount - firstIndex

Reply:
No, my suggestion is also not correct. There needs to be some more magic so that the sum of all mgn withdrawals adds up to totalMGN

Reply:
I think the current logic is correct. `totalPoolSharesCummulative` will be the integral of the entire step function from beginning to end. And the claimable share should have the same ratio to the total MGN available as the rectangle of the `deposit` * `duration` to the total integral.

Can you show an example where this is incorrect?

Reply:
Uh, I got the variable totalPoolSharesCummulative wrong. 

Now I got it. Makes perfect sense. And yes, all withdrawals add up to the correct number. 
sweet